java-squid/src/main/java/org/sonar/java/resolve/ParametrizedTypeCache.java

```diff
@@ -41,4 +46,18 @@ public JavaType getParametrizedTypeType(JavaSymbol.TypeJavaSymbol symbol, TypeSu
     return typeCache.get(symbol).get(typeSubstitution);
   }
 
+  public JavaType getWildcardType(JavaSymbol.TypeJavaSymbol bound, BoundType boundType) {
```

Comment:
bound should probably be the type and not the symbol. Moreover what about unknown bound ?